embabel-common-ai/src/main/kotlin/com/embabel/common/ai/converters/streaming/support/StreamingUtils.kt

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2024-2025 Embabel Software, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.embabel.common.ai.converters.streaming.support
+
+import com.embabel.common.core.streaming.ThinkingState
+import org.slf4j.LoggerFactory
+
+/**
+ * Utility functions for streaming content processing, particularly thinking content detection and extraction.
+ *
+ * Provides centralized logic for identifying and processing thinking content in various formats
+ * used by different LLM models and reasoning systems.
+ */
+internal object StreamingUtils {
```

Comment:
Except for `isValidJson` (see comment below), all of the methods in this utility type are related to detecting thinking content, so perhaps `ThinkingDetector` would be a better name for it.

By changing the name, it makes more sense to use this utility in non-streaming scenario's, eg. in the `SuppressThinkingConverter`, whereas without the name change we would have to import `StreamingUtils` in non-streaming code.